src/NuGet.Clients/PackageManagement.UI/PackageSolutionDetailControlModel.cs

```diff
@@ -412,6 +413,34 @@ private static bool IsInstalled(NuGetProject project, string id)
             }
         }
 
+        // auto select projects based on the current tab and currently selected package
+        private void AutoSelectProjects()
+        {
+            if (_filter == Filter.Consolidate ||
+                _filter == Filter.UpdatesAvailable)
+            {
+                foreach (var project in _projects)
+                {
+                    project.IsSelected = project.InstalledVersion != null;
+                }
+            }
+        }
+
+        public override void OnFilterChanged(Filter filter)
```

Comment:
we need to clear selects when the current tab is switched to "Browser" or "install".